File: ruby_aoc/2021/day01/day01.rb
require_relative '../helpers/input_reader'

execute(1) do |lines|
  lines.map(&:to_i).each_cons(2).count { |a, b| b > a }
end

execute(2) do |lines|
  lines.map(&:to_i).each_cons(3).map { |a, b, c| a + b + c }.each_cons(2).count { |a, b| b > a }
end

File: kotlin_aoc/src/main/kotlin/aoc2021/day01.kt
fun main() {
    day01()
}

fun day01() {
    execute("2021", "01", "1") { lines ->
        lines.map { it.toInt() }.windowed(2).count { (a, b) -> b > a }
    }
    execute("2021", "01", "2") { lines ->
        lines.map { it.toInt() }
            .windowed(3)
            .map { it.sum() }
            .windowed(2)
            .count { (a, b) -> b > a }
    }
}
